### PR TITLE
Alerting: Add new API endpoint GET /api/v1/ngalert

### LINKED ISSUE #51321: Alerting: Alertmanager configuration status endpoint
To proceed with https://github.com/grafana/grafana/issues/49212 we would need a new endpoint / augment an existing endpoint to figure out of Grafana has been configured to _only_ use an external Alertmanager.

### PR DESCRIPTION
**Pull request by yuri-tceretian**
**What this PR does / why we need it**:
When Grafana alerting user chooses to send Grafana-managed alerts to an external Alertmanager the information in the tab `Alert Group` (when Grafana source is selected) can be confusing because it reflects the status of the local Alertmanager that does not receive any alerts (due to the settings mentioned above). Also, for non-admin users that do not have access to the configuration that mentions the selection of the external Alertmanager the tabs like "Notification Policies", "Contact Points", "Silences" can be confusing because changes made in those tabs will not be applied to the external Alertmanager. 

Therefore, UI should notify users when Grafana alerts are sent to external Alertmanager. At this moment, there is no API endpoint that would be available to regular (non-admin) users and could provide the necessary information. This PR introduces a new endpoint `GET /api/v1/ngalert` that is available to any users that have any alerting read permissions. For now, it returns just a simple JSON object 
```
{
    "alertmanagersChoice": "internal"|"external"|"all"
}
```
and later can be expanded to provide some general information about the alerting engine (for example, available data sources that UI can query for alert rules)

Fixes: https://github.com/grafana/grafana/issues/51321

### Changelog
- Alerting: Introduced a new API endpoint `GET /api/v1/ngalert` that returns the status\configuration of the alerting engine

### CODE REVIEW

**Comment by grobinson-grafana** [comment]
I like it! Perhaps we should add something else to the URL just to give us more options later. For example: 

- `/api/v1/ngalert/alertmanagers`
- `/api/v1/ngalert/info`
- `/api/v1/ngalert/preferences`
- `/api/v1/ngalert/status`

**Review by grobinson-grafana — APPROVED** [review]

**Comment by grobinson-grafana** [comment]
Looks good to me, let's get approval from one other person in @grafana/alerting-squad.

**Comment by yuri-tceretian** [comment]
`/api/v1/ngalert/status` Initially, I went this way but later fixed it to be without `status` because, IMO, it's not RESTful: there is no entity "status", and there is no any other verbs that can be applied to it.

**Comment by grobinson-grafana** [comment]
> /api/v1/ngalert/status Initially, I went this way but later fixed it to be without status because, IMO, it's not RESTful: there is no entity "status", and there is no any other verbs that can be applied to it.

Can the same not be said about `/api/v1/ngalert`?

**Review by JohnnyQQQQ — APPROVED** [review]
LGTM, I think the endpoint is fine, even if it's not the "ngalert" resource.

**Review by alexweav — APPROVED** [review]
LGTM, nice. Straightforward and consistent with current endpoints.

**Comment by joeblubaugh** [comment]
What can we do to keep `ngalert` out of the API path? It won't be "next-gen" now that it's the default, and I'd hate to see that hang around forever.

File: pkg/services/ngalert/api/tooling/definitions/admin.go
Comment (joeblubaugh) [reply pkg/services/ngalert/api/tooling/definitions/admin.go:7]: I like this idea. I really don't want to have `ngalert` in a stable API if we can avoid it.

Comment (yuri-tceretian) [reply pkg/services/ngalert/api/tooling/definitions/admin.go:7]: +1 for alternative, i.e. making it stable when we switch to `alerting`

**Comment by yuri-tceretian** [comment]
> > /api/v1/ngalert/status Initially, I went this way but later fixed it to be without status because, IMO, it's not RESTful: there is no entity "status", and there is no any other verbs that can be applied to it.
> 
> Can the same not be said about `/api/v1/ngalert`?

In my mind, `ngalert` is a resource - the alerting engine. One of its properties - the current (configuration) or status. Status is not a resource, it's a property of one.

**Comment by grobinson-grafana** [comment]
> In my mind, `ngalert` is a resource - the alerting engine. One of its properties - the current (configuration) status. Status is not a resource, it's a property of one.

What about `/api/v1/ngalert/configuration`? I'm just not sure we should consider `ngalert` a resource for the same reasons `api` is not a resource.

**Comment by yuri-tceretian** [comment]
> > In my mind, `ngalert` is a resource - the alerting engine. One of its properties - the current (configuration) status. Status is not a resource, it's a property of one.
> 
> What about `/api/v1/ngalert/configuration`? I'm just not sure we should consider `ngalert` a resource for the same reasons `api` is not a resource.

If `api` can provide some meaningful information then I do not see why it can't be a resource. There is already a resource `admin_config` which is available to only Admin users and provides AlertmanagerChoice and a list of external Alertmanagers. So, it is focused on "configuration", if the configuration does not exist it returns 404, which is ok for configuration. The purpose of the endpoint I introduce is different: 1. it provides configuration from a different angle, if the config is not available it will return information anyway. 2. In the future it will be able to provide more information not related to configuration exactly. For example, it can provide information about available and healthy datasources that support alerting so UI can just query them without having to determine the status of each of them by itself, which causes the UI to be incredibly unresponsive in the case of a  big list of data sources to scan and problems with connectivity to the datasoruces.

**Comment by grobinson-grafana** [comment]
> it provides configuration from a different angle, if the config is not available it will return information anyway.

But then I don't understand the RESTful argument against `status` as RESTful APIs should return 404 if the config is not available? I'm not against this being merged as I have approved it, and we can change it if we need in future, I am just unconvinced that `ngalert` is the best URL for this :)

**Comment by yuri-tceretian** [comment]
> But then I don't understand the RESTful argument against status as RESTful APIs should return 404 if the config is not available? I'm not against this being merged as I have approved it, and we can change it if we need in future, I am just unconvinced that ngalert is the best URL for this :)

I am not in favor of `status` because it adds one more segment adjacent to other segments, which logically puts them on the same level as other "resources", like admin config, whereas it should aggregate information from the lower level resources.

**Pull request merged**
